spec/abstract/KHeap_A.thy

```diff
@@ -42,15 +48,19 @@ where
 section "TCBs"
 
 definition
-  get_tcb :: "obj_ref \<Rightarrow> 'z state \<Rightarrow> tcb option"
+  get_tcb :: "obj_ref \<Rightarrow> (tcb, 'z::state_ext ) r_monad"
 where
   "get_tcb tcb_ref state \<equiv>
-   case kheap state tcb_ref of
+   case read_object tcb_ref state of
```

Comment:
I would change this into a projection and separate reader like for `read_object`. 

Something like
```
definition
tcb_of obj == case obj of TCB tcb => Some tcb | _ => None

abbreviation
tcbs_of s == kheap s |> tcb_of

definition
get_tcb = swp tcbs_of
```

To not have to update all of the proofs, you could still define a lemma `get_tcb_def` as before, and maybe give the actual `get_tcb` definition the name `get_tcb_tcbs_of`.

We would then have the projection available for preconditions etc and have a direct connection to the definition.